### PR TITLE
Add enable ts for precompiled contracts

### PR DESCRIPTION
**Pull request by ninjaahhh**

### CODE REVIEW

File: quarkchain/evm/messages.py [hunk 1]
```diff
@@ -564,7 +564,11 @@ def _apply_msg(ext, msg, code):
 
     # Main loop
     if msg.code_address in ext.specials:
```
Comment (qizhou) [inline quarkchain/evm/messages.py]: Why not use

special_proc, enable_ts = ext.specials.get(msg.code_address, (None, 0))
if special_proc is not None and ext.block_timestamp >= enable_ts:
...

Comment (ninjaahhh) [reply quarkchain/evm/messages.py]: good point. done

File: quarkchain/env.py [hunk 2]
```diff
@@ -1,14 +1,16 @@
...
 from quarkchain.cluster.cluster_config import ClusterConfig
+from quarkchain.evm.specials import specials
 
 
 class Env:
-    def __init__(self, db=None, evm_config=None, cluster_config=None):
+    def __init__(self, db=None, evm_config=None):
         self.db = db or InMemoryDb()
-        self.cluster_config = cluster_config if cluster_config else ClusterConfig()
+        self.__cluster_config = ClusterConfig()
```
Comment (qizhou) [inline quarkchain/env.py]: Why not still use cluster_config and then apply timestamp in the ctor?  This makes sure we have the best backward-compatible interface.

Comment (ninjaahhh) [reply quarkchain/env.py]: some places we are manually setting the cluster config (like in `slave.py`):

```python
    env = DEFAULT_ENV.copy()
    env.cluster_config = ClusterConfig.create_from_args(args)
    env.slave_config = env.cluster_config.get_slave_config(args.node_id)
    env.arguments = args
```

Comment (qizhou) [reply quarkchain/env.py]: Then I would suggest using

self.cluster_config = ClusterConfig()

**Pull request merged**